### DISCUSSION TITLE
LIghtWeigth Richtext control with clickable content

### DISCUSSION THREAD

**Discussion by lucaregini** [unoplatform/uno#11744] [Q&A]
I need to have a control that supports a scrollable blob of text. The text displayed in there should be styled in 4 possible different ways depending on the specific content. I must be able to identify any "word" that the user has clicked on. At the same time I should be able to programmatically scroll to any of the words within the blob of text based on actions made on other controls in the same window.
So far I haven't been able to find a way to achieve that a part from embedding a webview. Is it really my only option?

I have no images or other Rich text elements. Only clickable text with 4 possible different colours and size.

**Comment by jeromelaban** [comment]
Thanks for the report. At this time, we do not a WinUI-based API for RichText control (https://github.com/unoplatform/uno/issues/3848). You can try using javascript-based controls directly without an iframe by using [javascript interop](https://platform.uno/docs/articles/interop/wasm-javascript-1.html).
